core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableSchemaBuilder.java

```diff
@@ -224,7 +224,7 @@ private ColumnSchema addColumn(StructField field, String parentName, AtomicInteg
         }
       }
     }
-    if (newColumn.isDimensionColumn()) {
+    if (newColumn.isDimensionColumn() && newColumn.isSortColumn()) {
```

Comment:
@jackylk In case of no sort , isSortColumn will be true for all the dimension columns. so above check is of no use.
So I suggest to do like below.
In case of no sort for each dimension column we have set isSortColumn to false  and keep your condition

Reply:
When I change CarbonCliTest to write carbon files with no sort, 
```
TestUtil.writeFilesAndVerify(5000000, new Schema(fields), path, new String[0], 3, 8);
```
The output of the CarbonCli is
```
## Schema
schema in part-0-31810056985663_batchno0-0-null-31809010599854.carbondata
version: V3
timestamp: 2018-09-29 09:41:19.839
Column Name  Data Type  Column Type  SortColumn  Encoding  Ordinal  Id  
name         STRING     dimension    false       []        0        NA  
age          INT        measure      false       []        1        NA  
```
So it indicate that the sort columns property in each ColumnSchema is false, right?